lxd/daemon.go

```diff
@@ -334,12 +334,30 @@ func (d *Daemon) Authenticate(w http.ResponseWriter, r *http.Request) (trusted b
 	}
 
 	if d.oidcVerifier != nil && d.oidcVerifier.IsRequest(r) {
-		userName, err := d.oidcVerifier.Auth(d.shutdownCtx, w, r)
+		result, err := d.oidcVerifier.Auth(d.shutdownCtx, w, r)
 		if err != nil {
 			return false, "", "", err
 		}
 
-		return true, userName, api.AuthenticationMethodOIDC, nil
+		_, err = d.identityCache.Get(api.AuthenticationMethodOIDC, result.Subject)
+		if err != nil {
```

Comment:
I dont think we should be considering all errors from the .Get call to mean "missing identity", instead we should be inspection the error for a NotFound status error.